pincer/objects/guild/channel.py

```diff
@@ -661,6 +662,347 @@ async def create_invite(
+                    "type": type_,
+                    "invitable": invitable,
+                    "rate_limit_per_user": rate_limit_per_user
+                }
+            )
+        ))
+
+    async def join_thread(self):
+        """
+        Adds the current user to a thread.
+        Also requires the thread to not be archived.
+        """
+        await self._http.put(f"channels/{self.id}/thread-members/@me")
+
+    async def add_thread_member(self, user: User):
+        """
+        Adds another member to a thread.
+        Requires the ability to send messages in the thread.
+        Also requires the thread to not be archived.
+
+        Parameters
+        ----------
+        user: :class:`~.pincer.objects.user.User`
+            The user to add to the thread.
+        """
+        await self._http.put(f"channels/{self.id}/thread-members/{user.id}")
+
+    async def leave_thread(self):
+        """
+        Removes the current user from a thread.
+        Also requires the thread to not be archived.
+        """
+        await self._http.delete(f"channels/{self.id}/thread-members/@me")
+
+    async def remove_thread_member(self, user: User):
+        """
+        Removes another member from a thread. Requires the ``MANAGE_THREADS``
+        permission, or the creator of the thread if it is a
+        ``GUILD_PRIVATE_THREAD``. Also requires the thread to not be archived.
+
+        Parameters
+        ----------
+        user: :class:`~.pincer.objects.user.User`
+            The user to remove from the thread.
+        """
+        await self._http.delete(f"channels/{self.id}/thread-members/{user.id}")
+
+    async def get_thread_member(self, user: User) -> ThreadMember:
+        """
+        Returns a thread member object for the specified user if they are a
+        member of the thread.
+
+        Parameters
+        ----------
+        user: :class:`~.pincer.objects.user.User`
+            The user to get the thread member for.
+
+        Returns
+        -------
+        :class:`~.pincer.objects.channel.ThreadMember`
+            The thread member object.
+        """
+        return ThreadMember.from_dict(construct_client_dict(self._client,
+            await self._http.get(f"channels/{self.id}/thread-members/{user.id}")
+        ))
+
+    async def list_thread_members(self) -> AsyncIterator[ThreadMember]:
+        """
+        Fetches all the thread members for the thread. Returns an iterator of
+        ThreadMember objects.
+
+        Returns
+        -------
+        AsyncIterator[:class:`~.pincer.objects.channel.ThreadMember`]
+            An iterator of thread members.
+        """
+        data = await self._http.get(f"channels/{self.id}/thread-members")
+        for member in data:
+            yield ThreadMember.from_dict(
+                construct_client_dict(self._client, member)
+            )
+
+    async def list_active_threads(self) -> dict[str, Any]:
```

Comment:
Python 3.8 doesn't allow typehinting generics
```suggestion
    async def list_active_threads(self) -> Dict[str, Any]:
```